example/googleMapsExample.js

```diff
@@ -0,0 +1,605 @@
+
+	tiles.manager.addHandler( /\.gltf$/, loader );
+	offsetParent.add( tiles.group );
+
+	tiles.setResolutionFromRenderer( camera, renderer );
+	tiles.setCamera( camera );
+
+}
+
+function isInt( input ) {
+
+	return ( typeof input === 'string' ) ? ! isNaN( input ) && ! isNaN( parseFloat( input ) ) && Number.isInteger( parseFloat( input ) ) : Number.isInteger( input );
+
+}
+
+function reinstantiateTiles() {
+
+	let url = hashUrl || '../data/tileset.json';
+
+	if ( hashUrl ) {
+
+		params.ionAssetId = isInt( hashUrl ) ? hashUrl : '';
+
+	}
+
+	if ( tiles ) {
+
+		offsetParent.remove( tiles.group );
+		tiles.dispose();
+		tiles = null;
+
+	}
+
+	url = new URL( `${apiOrigin}/v1/3dtiles/root.json?key=${ params.apiKey }` );
+
+	fetch( url, { mode: 'cors' } )
+		.then( ( res ) => {
+
+			if ( res.ok ) {
+
+				return res.json();
+
+			} else {
+
+				return Promise.reject( `${res.status} : ${res.statusText}` );
+
+			}
+
+		} )
+		.then( ( json ) => {
+
+			if ( !json.root ) {
+				throw new Error( `malformed response: ${ json }` );
+			}
+
+			// TODO: check this is correct
+			// find first node with uri and treat that as root
+			let uri = undefined;
+			const toVisit = [];
+			for ( let curr = json.root; curr !== undefined; curr = toVisit.pop() ) {
+				if ( curr.content?.uri ) {
+					uri = new URL( `${ apiOrigin }${ curr.content.uri }` );
+					uri.searchParams.append( 'key', params.apiKey );
+					break;
+				}
+
+				toVisit.push( ...curr.children );
+			}
+
+			if ( !uri ) {
+				throw new Error( `can't find session string in response: ${ json }` );
+			}
+
+			const session = uri.searchParams.get( 'session' );
+			console.log('session', session);
+
+			tiles = new TilesRenderer( url.toString() );
+
+			tiles.preprocessURL = uri => {
+
+				uri = new URL( uri );
+				if ( /^http/.test( uri.protocol ) ) {
+
+					uri.searchParams.append( 'session', session );
+					uri.searchParams.append( 'key', params.apiKey );
+
+				}
+				return uri.toString();
+
+			};
+
+			tiles.onLoadTileSet = () => {
+
+				const box = new Box3();
+				const sphere = new Sphere();
+				const matrix = new Matrix4();
+
+				let position;
+				let distanceToEllipsoidCenter;
+
+				if ( tiles.getOrientedBounds( box, matrix ) ) {
+
+					position = new Vector3().setFromMatrixPosition( matrix );
+					distanceToEllipsoidCenter = position.length();
+
+				} else if ( tiles.getBoundingSphere( sphere ) ) {
+
+					position = sphere.center.clone();
+					distanceToEllipsoidCenter = position.length();
+
+				}
+
+				const surfaceDirection = position.normalize();
+				const up = new Vector3( 0, 1, 0 );
+				const rotationToNorthPole = rotationBetweenDirections( surfaceDirection, up );
+
+				tiles.group.quaternion.x = rotationToNorthPole.x;
+				tiles.group.quaternion.y = rotationToNorthPole.y;
+				tiles.group.quaternion.z = rotationToNorthPole.z;
+				tiles.group.quaternion.w = rotationToNorthPole.w;
+
+				tiles.group.position.y = - distanceToEllipsoidCenter;
+
+			};
```

Comment:
What is this callback implementation for?

Reply:
I'm adapting your cesium sample while I get my head around the API / the 3Dtile format. I might not have pulled out everything non-essential.

Reply:
That's in part what I was thinking, as well. I think that should help.